Changelog 140 (#105)
intellij-rust/intellij-rust.github.io: _posts/2021-02-01-changelog-140.markdown

  `#![cfg_attr(feature = "nightly", feature(generators))]`

<!-- https://github.com/intellij-rust/intellij-rust/pull/6691 -->
* Improve package completion in `Cargo.toml` (by [@Pzixel])

<!-- https://github.com/intellij-rust/intellij-rust/pull/6668 -->
* Fix `Join Lines` action (<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>J</kbd>):
  in the case when there isn't a space between parameters (by [@t-kameyama])

<!-- https://github.com/intellij-rust/intellij-rust/pull/6684 -->
<!-- https://github.com/intellij-rust/intellij-rust/pull/6678 -->
* Extract only selected statements in `Extract function` refactoring (by [@Kobzol])

<!-- https://github.com/intellij-rust/intellij-rust/pull/6609 -->
* Download stdlib dependencies into IDE system directory instead of `src`
  directory in toolchain location to avoid unexpected behavior like read-only directory.
  Note, stdlib dependencies are downloaded by the plugin only when
  `org.rust.cargo.fetch.actual.stdlib.metadata` experimental feature is enabled,
  otherwise the plugin uses hardcoded stdlib structure

<!-- https://github.com/intellij-rust/intellij-rust/pull/6580 -->

> * Fix `Change item visibility` intention so that it is not offered in trait impl blocks. (by [@Kobzol])

Reviewer: ```suggestion
* Do not suggest `Change item visibility` intention in trait impl blocks (by [@Kobzol])
```

---
layout: post
title: "IntelliJ Rust Changelog #140"
date: 2021-02-01 17:00:00 +0300
---

This is the last release that supports the 2020.2 platform. To receive plugin updates further, please upgrade your IDE to 2020.3.

## New Features

<!-- https://github.com/intellij-rust/intellij-rust/pull/5944 -->
* Add `Change Signature` refactoring (`Refactor | Change Signature` or <kbd>Ctrl</kbd>+<kbd>F6</kbd>)
  for functions and methods (by [@Kobzol])

  {% include gif-img.html path="/assets/posts/changelog-140/5944" w="700px" %}

<!-- https://github.com/intellij-rust/intellij-rust/pull/6563 -->
* Introduce `Share in Playground` action to share your code on https://play.rust-lang.org/.
  You can invoke it via `Tools | Rust | Share in Playground` or via context menu

>   {% include gif-img.html path="/assets/posts/changelog-140/6563" w="700px" %}

Reviewer 2: Let's create a new gif where confirmation dialog is shown as well
